packages/compartment-mapper/src/node-modules.js

```diff
@@ -1048,6 +1026,93 @@ export const compartmentMapForNodeModules = async (
   return compartmentMap;
 };
 
+/**
+ * @param {ReadFn | ReadPowers | MaybeReadPowers} readPowers
+ * @param {Array<{packageLocation:string, packageDescriptor:PackageDescriptor, moduleSpecifier:string}>} entriesToProcess
+ * @param {Set<string>} conditionsOption
+ * @param {CompartmentMapForNodeModulesOptions} [options]
+ * @returns {Promise<CompartmentMapDescriptor>}
+ * @deprecated Use {@link mapNodeModules} instead.
+ */
+export const compartmentMapForNodeModules2 = async (
+  readPowers,
+  entriesToProcess,
+  conditionsOption,
+  options = {},
+) => {
+  await null;
+  const {
+    dev = false,
+    commonDependencies = {},
+    policy,
+    strict = false,
+    log = noop,
+  } = options;
+  const { maybeRead, canonical } = unpackReadPowers(readPowers);
+  const languageOptions = makeLanguageOptions(options);
+
+  const conditions = new Set(conditionsOption || []);
+
+  // dev is only set for the entry package, and implied by the development
+  // condition.
+  // The dev option is deprecated in favor of using conditions, since that
+  // covers more intentional behaviors of the development mode.
+
+  const memo = create(null);
+  const graph = create(null);
+  const preferredPackageLogicalPathMap = new Map();
+
+  const firstEntry = entriesToProcess[0];
+
+  for (const entrypoint of entriesToProcess) {
+    const { packageLocation, packageDescriptor, moduleSpecifier } = entrypoint;
+    const logicalPath =
+      entrypoint === firstEntry
+        ? []
+        : [`$external:${packageDescriptor.name || packageLocation}`];
```

Comment:
Not super happy with this yet. The discrepancy between item in path and package name might be getting in the way. 
I've considered overriding `packageDescriptor.name` for compartment map. Not sure what consequences that would have.